ndcube/tests/test_ndcube.py

```diff
@@ -837,7 +837,7 @@ def test_world_to_pixel(test_input, expected):
 
 
 @pytest.mark.parametrize("test_input,expected", [
-    ((cubem, [0.7*u.deg, 1.3e-5*u.deg, 1.02e-9*u.m], [1*u.deg, 1*u.deg, 1.06*u.m]), cubem[:, :2]),
+    ((cubem, [0.7*u.deg, 1.3e-5*u.deg, 1.02e-9*u.m], [1*u.deg, 1*u.deg, 5.e-11*u.m]), cubem[:, :, :3]),
```

Comment:
Can you add a couple more test cases here to use the ```upper_corner``` input, rather than just ```interval_widths```?  To do this, you will have to add a third element to the existing test_input tuples of ```None```, i.e. ```(cubem, [0.7*u.deg, 1.3e-5*u.deg, 1.02e-9*u.m], [1*u.deg, 1*u.deg, 5.e-11*u.m], None)```.  In order to use ```upper_corner``` in the new cases, the second element of the new tuples will be ```None``` (```interval_widths```) while the third element will be the ```upper_corner``` value.

In addition, we need a whole new test, ```test_crop_by_coords_error```, which tests the correct errors are raised by the correct combination of inputs.  This is an easier test function to write as you don't have to work out the output.  See here for an example of such a test: https://github.com/sunpy/ndcube/blob/master/ndcube/tests/test_utils_wcs.py#L80

Reply:
Ok, I will check that.